src/lib389/lib389/cli_base/dsrc.py

```diff
@@ -145,16 +145,18 @@ def dsrc_to_ldap(path, instance_name, log):
 
     dsrc_inst['tls_cert'] = config.get(instance_name, 'tls_cert', fallback=None)
     dsrc_inst['tls_key'] = config.get(instance_name, 'tls_key', fallback=None)
-    dsrc_inst['tls_reqcert'] = config.get(instance_name, 'tls_reqcert', fallback='hard')
-    if dsrc_inst['tls_reqcert'] not in ['never', 'allow', 'hard']:
-        raise Exception("dsrc tls_reqcert value invalid. %s [%s] tls_reqcert should be one of never, allow or hard" % (instance_name,
-                                                                                                                       path))
+    dsrc_inst['tls_reqcert'] = config.get(instance_name, 'tls_reqcert', fallback=None)
     if dsrc_inst['tls_reqcert'] == 'never':
         dsrc_inst['tls_reqcert'] = ldap.OPT_X_TLS_NEVER
     elif dsrc_inst['tls_reqcert'] == 'allow':
         dsrc_inst['tls_reqcert'] = ldap.OPT_X_TLS_ALLOW
-    else:
+    elif dsrc_inst['tls_reqcert'] == 'hard':
         dsrc_inst['tls_reqcert'] = ldap.OPT_X_TLS_HARD
+    elif dsrc_inst['tls_reqcert'] is None:
+        # Use system value
+        pass
+    else:
+        raise Exception("dsrc tls_reqcert value invalid. %s [%s] tls_reqcert should be one of never, allow or hard" % (instance_name, path))
```

Comment:
Could you make this exception a ValueError()?  It will help with CLI and UI error handling

Reply:
Otherwise ACK!